pytorch_lightning/trainer/training_loop.py

```diff
@@ -675,6 +698,8 @@ def run_training_teardown(self):
         # summarize profile results
         self.profiler.describe()
 
+        self._teardown_already_run = True
```

Comment:
just wondering, could this not also be moved into a decorator? It seems this variable is only used here and not somewhere else in the trainer.

Reply:
It is and thus it's initialized just here. I tried to avoid more complicated stuff for now, since I'm not that familiar with all the signal handling. But probably we can do it

Reply:
ok sounds good

Reply:
I guess that the trainer state would solve all these bool variables